10up/gutenbridge
# fix/170: wait for DOM manipulations before saving post

### Description of the Change


The issue happens because when images are added via replaceBlocks(), it takes some time for the browser to first inject `<figure />` and then append `<img />` to the figure, and since this is an asynchronous process, the client.save() triggers before all the DOM manipulations inside the blocks are completed.

This PR implements MutationObserver to detect if there are any DOM modifications within the replaced block and waits until all DOM are done before saving the post.


Closes #170 

### How to test the Change


1. Activate classic editor.
2. Create 2 posts with gallery and add `heading`. (Just to test if the PR works as expected to transform other elements to blocks).
3. Disable the classic editor.
4. Run the following: `wp convert-to-blocks start --post_type=post`
5. Open the URL displayed in the terminal.
6. It will start converting blocks in both the posts.
7. Wait until terminal shows the message: "Migration finished successfully."
8. Open both the posts and reload the page to verify if the transformed blocks are visible as expected.

### Changelog Entry


> Fixed - Issue with saving post before the convert to blocks transform completed.


### Credits

Props @mdesplenter @Sidsector9 @dsawardekar 


### Checklist:
